File: platform/polyglot/confluence_api/py/src/confluence_api/services/label_service.py
"""
Label service for Confluence Data Center REST API v9.2.3.

Provides operations for retrieving related labels and recently used labels.
"""

from __future__ import annotations

from typing import Any

from confluence_api.logger import create_logger

log = create_logger("confluence_api", __file__)


class LabelService:
    """Service for Confluence label operations."""

    def __init__(self, client: Any) -> None:
        self._client = client

    def get_related_labels(self, label_name: str) -> dict:
        """Retrieve labels related to a given label."""
        log.debug("get_related_labels called", {"label_name": label_name})
        result = self._client.get(f"label/{label_name}/related")
        log.info("get_related_labels succeeded", {"label_name": label_name})
        return result

    def get_recent_labels(self) -> dict:
        """Retrieve recently used labels."""
        log.debug("get_recent_labels called")
        result = self._client.get("label/recent")
        log.info("get_recent_labels succeeded")
        return result


class AsyncLabelService:
    """Async service for Confluence label operations."""

    def __init__(self, client: Any) -> None:
        self._client = client

    async def get_related_labels(self, label_name: str) -> dict:
        """Retrieve labels related to a given label."""
        log.debug("get_related_labels called", {"label_name": label_name})
        result = await self._client.get(f"label/{label_name}/related")
        log.info("get_related_labels succeeded", {"label_name": label_name})
        return result

    async def get_recent_labels(self) -> dict:
        """Retrieve recently used labels."""
        log.debug("get_recent_labels called")
        result = await self._client.get("label/recent")
        log.info("get_recent_labels succeeded")
        return result

File: platform/polyglot/confluence_api/mjs/src/services/label-service.mjs
/**
 * @module services/label-service
 * @description Service for Confluence Label REST API operations.
 *
 * Provides access to global label relationships and recently-used labels.
 *
 * Confluence Data Center REST API v9.2.3.
 */

import { createLogger } from '../logger.mjs';

const log = createLogger('confluence_api', import.meta.url);

/**
 * Service class for Confluence label operations.
 */
export class LabelService {
  /**
   * @param {import('../client/FetchClient.mjs').FetchClient} client
   *   The Confluence fetch client instance.
   */
  constructor(client) {
    /** @private */
    this._client = client;
  }

  /**
   * Get labels related to a given label.
   *
   * Returns labels that frequently co-occur with the specified label
   * across content items.
   *
   * @param {string} labelName - The label name to find related labels for.
   * @returns {Promise<Object>} Related labels.
   */
  async getRelatedLabels(labelName) {
    log.debug('getRelatedLabels called', { labelName });
    try {
      const result = await this._client.get(`label/${labelName}/related`);
      log.info('getRelatedLabels succeeded', { labelName });
      return result;
    } catch (error) {
      log.error('getRelatedLabels failed', { labelName, message: error.message, status: error.status });
      throw error;
    }
  }

  /**
   * Get recently-used labels.
   *
   * Returns labels that have been recently applied to content by the
   * current user or across the instance.
   *
   * @returns {Promise<Object>} Recent labels.
   */
  async getRecentLabels() {
    log.debug('getRecentLabels called');
    try {
      const result = await this._client.get('label/recent');
      log.info('getRecentLabels succeeded');
      return result;
    } catch (error) {
      log.error('getRecentLabels failed', { message: error.message, status: error.status });
      throw error;
    }
  }
}
